dht/dht.go

```diff
@@ -52,6 +66,8 @@ type ServerConfig struct {
 	IPBlocklist iplist.Ranger
 	// Used to secure the server's ID. Defaults to the Conn's LocalAddr().
 	PublicIP net.IP
+	// Hooks for the server to call on various KRPC queries
+	KRPCHooks map[string]KRPCHook
```

Comment:
I feel this doesn't reflect what it actually does. Further, the "hook" function can work out what the query is for itself, since it receive the message. I think this should simply be OnQuery func(*Msg, net.Addr) bool. With a note that the Msg will be processed with any changes made if the function returns true. It's much clearer.